cmd/deploy/deploy.go

```diff
@@ -91,7 +92,23 @@ func Deploy(ctx context.Context) *cobra.Command {
 			// This is needed because the deploy command needs the original kubeconfig configuration even in the execution within another
 			// deploy command. If not, we could be proxying a proxy and we would be applying the incorrect deployed-by label
 			os.Setenv(model.OktetoWithinDeployCommandContextEnvVar, "false")
-			if err := contextCMD.Run(ctx, &contextCMD.ContextOptions{}); err != nil {
+			if okteto.IsOkteto() && options.Namespace != "" {
+				create, err := utils.ShouldCreateNamespace(ctx, options.Namespace)
```

Comment:
For this case doesn't matter because pipeline installer is not specifying any namespace when using `okteto deploy` command but we have to bear in mind for the future that the command is being used in a non-interactive way in the installer